components/coding/profileV2/achievement_components/AchievementComponent.tsx

```diff
@@ -13,18 +13,21 @@ import {
 } from "../../../../graphql/coding/userBadges/updateUserCodingBadges";
 import UnitBadgeSection from "./UnitBadgeSection";
 import findBadgeDiff from "./findBadgeDiff";
+import {
+  FetchUserRoleData,
+  FETCH_USER_ROLE,
+} from "../../../../graphql/fetchUserRole";
 
 export type AchievementComponentProps = {
   userId: string;
-  isEditable: boolean;
 };
 
-const AchievementComponent = ({
-  userId,
-  isEditable,
-}: AchievementComponentProps) => {
+const AchievementComponent = ({ userId }: AchievementComponentProps) => {
   const [unitBadges, setUnitBadges] = useState<IntroCourseUnit[]>();
+  // editMode set to true only when a coach accesses the page
   const [editMode, setEditMode] = useState(false);
+  // isEditable set to true only when a coach clicks the PencilAltIcon Component
+  const [isEditable, setIsEditable] = useState(false);
```

Comment:
I might rename this to say `isEditButtonVisible` so the edit button is only visible for coaches. And coaches have to click the edit button in order to turn on edit mode. 

Eventually, a coach should be able to navigate to a badge detail page by clicking on a badge when editMode is off.